osbs/build/build_request.py

```diff
@@ -465,21 +467,63 @@ def set_kerberos_auth(self, plugins):
             self.dj.dock_json_set_arg(phase, plugin, 'koji_principal', krb_principal)
             self.dj.dock_json_set_arg(phase, plugin, 'koji_keytab', krb_keytab)
 
-    def set_reactor_config_map(self):
-        if not self.spec.reactor_config_map.value:
+    def set_reactor_config(self):
+        if not self.spec.reactor_config_map.value and not self.spec.reactor_config_override.value:
             return
         custom = self.template['spec']['strategy']['customStrategy']
 
-        reactor_config_map = {
-            'name': 'REACTOR_CONFIG',
-            'valueFrom': {
-                'configMapKeyRef': {
-                    'name': self.spec.reactor_config_map.value,
-                    'key': 'config.yaml'
+        if self.spec.reactor_config_override.value:
+            reactor_config = {
+                'name': 'REACTOR_CONFIG',
+                'value': yaml.dump(self.spec.reactor_config_override.value)
+            }
+        elif self.spec.reactor_config_map.value:
+            reactor_config = {
+                'name': 'REACTOR_CONFIG',
+                'valueFrom': {
+                    'configMapKeyRef': {
+                        'name': self.spec.reactor_config_map.value,
+                        'key': 'config.yaml'
+                    }
                 }
             }
-        }
-        custom['env'].append(reactor_config_map)
+
+        custom['env'].append(reactor_config)
+
+    def set_required_secrets(self):
+        """
+        Sets required secrets for worker build
+        """
+        if self.spec.platforms.value is not None:
+            return
+        if not self.spec.reactor_config_override.value:
+            return
+
+        data =self.spec.reactor_config_override.value
+        required_secrets = data.get('required_secrets')
+        if not required_secrets :
+            return
+
+        if 'secrets' not in self.template['spec']['strategy']['customStrategy']:
+            self.template['spec']['strategy']['customStrategy']['secrets'] = []
+
+        custom = self.template['spec']['strategy']['customStrategy']
```

Comment:
Nit: assign the custom first, then replace the if secrets block with:
custom.setdefault('secrets', [])

Reply:
`setdefault` also returns the value after it operates on it, so you could:

```python
secrets = self.template['spec']['strategy']['customStrategy'].setdefault('secrets')
```

That will return a reference to the `secrets` list which you can manipulate.

Reply:
ok